ocs_ci/ocs/must_gather.py

```diff
@@ -0,0 +1,119 @@
+import os
+import logging
+import shutil
+import tempfile
+
+from pathlib import Path
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.utils import collect_ocs_logs
+from ocs_ci.ocs.const_must_gather import must_gather_files
+
+
+logger = logging.getLogger(__name__)
+
+
+class MustGather(object):
+    """
+    MustGather Class
+    """
+    def __init__(self):
+        self.root = None
+        self.pod_path = None
+        self.temp_folder = None
+        self.empty_files = list()
+        self.files_not_exist = list()
+
+    def collect_must_gather(self):
+        """
+        Collect ocs_must_gather and copy the logs to a temporary folder.
+
+        """
+        self.temp_folder = tempfile.mkdtemp()
+        collect_ocs_logs(dir_name=self.temp_folder, ocp=False)
+        self.temp_folder += '_ocs_logs'
+        directory = os.path.join(self.temp_folder, 'ocs_must_gather')
+        for i in range(2):
+            folder = os.listdir(directory)
+            self.root = os.path.join(directory, folder[0])
+
+    def validate_file_size(self):
+        """
+        Validate the file is not empty
+
+        """
+        for path, subdirs, files in os.walk(self.root):
+            for file in files:
+                file_path = os.path.join(path, file)
+                if Path(file_path).stat().st_size == 0:
+                    logger.error(f"log file {file} empty!")
+                    self.empty_files.append(file)
+
+    def validate_expected_files(self):
```

Comment:
There are difference between versions. for example, in 4.5 there is more commands than earlier versions, in 4.6 directories structure is different than earlier versions - how do you handle this?